DSCResources/cChocoPackageInstall/cChocoPackageInstall.psm1

```diff
@@ -96,10 +98,25 @@ function Set-TargetResource
             $whatIfShouldProcess = $pscmdlet.ShouldProcess("$Name", 'Installing / upgrading package from Chocolatey')
             if ($whatIfShouldProcess) {
                 if ($Version) {
-                    Write-Verbose -Message "Uninstalling $Name due to version mis-match"
-                    UninstallPackage -pName $Name -pParams $Params
-                    Write-Verbose -Message "Re-Installing $Name with correct version $version"
-                    InstallPackage -pName $Name -pParams $Params -pVersion $Version -pSource $Source -cParams $chocoParams
+                    # Version check on the existing package to ensure we don't attempt to upgrade in a downgrade situation
+                    $installedPackages = @(Get-ChocoInstalledPackage | Where-object { $_.Name -eq $Name })
+                    $packageCount = $installedPackages.Count
+                    Write-Verbose "Found $packageCount matching package(s) for upgrade"
+                    $canUpgrade = $packageCount -eq 1 
+                    if ($canUpgrade) {
+                        [version] $installedVersion = $installedPackages[0].Version

```

Comment:
`[Version]` doesn't work for packages that are pre-release (for example a version of `2.11.1784.3-snapshot`.

Reply:
Good point - switched to using a regex to break apart the version